opsdroid/parsers/regex.py

```diff
@@ -32,7 +32,7 @@
                         "score": await calculate_score(
                             opts["expression"], opts["score_factor"]),
                         "skill": skill,
-                        "config": skill.config,
+                        "config": getattr(skill, 'config', getattr(skill, '__skill_attrs__', {}).get('config', None)),
```

Comment:
Could you explain a little what is going on here?

Reply:
Unfortunately, you can’t modify a bound method’s attributes (in case of a class-based skill, `skill` here is a bound method of the instance).

So i try get the `config` attribute of `skill` with `getattr` (equals to `skill.config`) and if that fails, i try to get the `__skill_attrs__` of the function (which is set by `Skill.set_matcher` and updated by `Skill.__init__` with the actual config).

Reply:
Yes, but at this specific place you don’t know the instance, only the instance method that was decorated. And while you can get the relevant class method from an instance method, you can’t get neither the instance nor the class object from that.

Reply:
And it immediately turned out i was wrong; `skill.__self__` holds the instance *if* the skill is from a class.  Let me rework this thing :)

Reply:
Ah you beat me to it! I just suggested this in the review below.